PROVISIONING-METAL.md

```diff
@@ -105,7 +105,30 @@ Please keep in mind that when using static addresses, DNS information must be su
   * `via` (IP address): Gateway IP address.  If no gateway is provided, a scope of `link` is assumed.
   * `route-metric` (integer): Relative route priority.
 
-Example `net.toml` with comments:
+Version `3` adds in support for bonding and vlan tagging. The support is limited to mode `1` (`active-backup`) for [bonding](https://www.kernel.org/doc/Documentation/networking/bonding.txt). Future support may include other bonding options - pull requests are welcome! Version `3` also adds the concept of network devices vs network interfaces. Physical devices like ethernet adapters are considered "interfaces", and network devices encompass other kinds of devices like bonds and vlans. This is a *breaking change* from version 2 and requires the user to specify 
+
+Migrating to version `3` from version `2` requires changes to how devices are defined. For example, if `[eno1]` is the current device name, changing a config from `[eno1]` to `[interface.eno1]` is all that is required. 
+
+Bonding configuration is created via a network device (`device`):
+* `bond` (map): To create a `bond` network device, specify `kind = "bond"`.
+  * `interfaces` (list of quoted strings of interfaces): Which interfaces should be added to the bond (i.e. `["eno1"]`). The first in the list is considered the default `primary`.
+  * `mode` (string): Currently `active-backup` is the only supported option
+  * `min-links` (int): Number of links required to bring up the device
+  * `monitoring` (map): Value must all be of `miimon` or `arpmon` type. The user must choose one type of monitoring and configure it fully in order for the bond to properly function. See [section 7](https://www.kernel.org/doc/Documentation/networking/bonding.txt) for more background on what to choose.
+    * `miimon-frequency` (int in ms): MII Monitoring frequency in miliseconds
+    * `miimon-updelay` (int in ms): MII Monitoring delay before the link is enabled after link is detected in milliseconds
+    * `miimon-downdelay` (int in ms): MII Monitoring delay before the link is disabled after link is no longer detected in milliseconds
+    * `arpmon-interval` (int in seconds): Number of seconds between intervals to determine link status, must be greater than 0
+    * `arpmon-validate` (enum of `all`, `none`, `active`, `backup`): What packets should be used to validate link
+    * `arpmon-targets` (list of quoted IPv4 address including prefix): List of targets to use for validating ARP. Min = 1, Max = 16
+
+Vlan tagging is configured as a new network device (`device`) stacked on another device:
+* `vlan` (map): Define by `kind = "vlan"`
+  * `device` (string for device): Defines the device the vlan should be configured on. If VLAN tagging is required, this device should recieve all IP address configuration instead of the underlying device.
+  * `id` (int): Number between 0 and 4096 specifying the vlan tag on the device
+
+
```

Comment:
nit: extra blank line

Again, just a markdown lint concern, should we choose to go that route.